validators/constructor.validator.js

```diff
@@ -0,0 +1,25 @@
+const Joi = require('joi');
+
+const { userNameRegExp } = require('../consts/regexp');
+
+const constructorValidator = Joi.object({
+  name: Joi.array().items({
+    lang: Joi.string()
+      .trim()
+      .required(),
+    value: Joi.string()
+      .trim()
+      .regex(userNameRegExp),
+  }),
+  material: Joi.object().required(),
+  color: Joi.object().required(),
+  image: Joi.string().required(),
+  basePrice: Joi.array().items(
+    { currency: Joi.string() },
+    { value: Joi.string() }
+  ),
+});
+
+module.exports = {
+  constructorValidator,
```

Comment:
your validation schema does not meet with graphQL input
<img width="287" alt="Screenshot 2021-03-23 at 18 01 42" src="https://user-images.githubusercontent.com/68334137/112178144-37dd3800-8c02-11eb-818c-a3ac3a41d8cb.png">
<img width="290" alt="Screenshot 2021-03-23 at 18 03 40" src="https://user-images.githubusercontent.com/68334137/112178503-770b8900-8c02-11eb-8056-4c93f4f824f8.png">
<img width="286" alt="Screenshot 2021-03-23 at 18 03 53" src="https://user-images.githubusercontent.com/68334137/112178546-7f63c400-8c02-11eb-83b2-0da8c12a2bd6.png">